ApplicationLibCode/ProjectDataModel/CellFilters/RimPolygonFilter.cpp

```diff
@@ -877,3 +867,64 @@ cvf::ref<RigPolyLinesData> RimPolygonFilter::polyLinesData() const
 
     return pld;
 }
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimPolygonFilter::initializeCellMap()
+{
+    for ( auto& i : m_cells )
+    {
+        delete i.second;
+    }
+    m_cells.clear();
+
+    int gridCount = RigReservoirGridTools::gridCount( m_srcCase() );
+    for ( int i = 0; i < gridCount; i++ )
+    {
+        m_cells[i] = new std::vector<size_t>();
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+/// Find which K layer we hit, in any of the grids, for any of the selected points
+//--------------------------------------------------------------------------------------------------
+int RimPolygonFilter::findEclipseKLayer( const std::vector<cvf::Vec3d>& points, RigEclipseCaseData* data )
+{
+    size_t ni, nj, nk;
+
+    // loop over all grids to find one with a cell hit
+    for ( size_t gridIndex = 0; gridIndex < data->gridCount(); gridIndex++ )
+    {
+        auto grid = data->grid( gridIndex );
+
+        // loop over all cells to find the correct one
```

Comment:
Consider using RigMainGrid::findIntersectingCells() to quickly find candidate cells to investigate.